### PR TITLE
Flash refresh + Refactor

### PR DESCRIPTION
**Pull request by emplums**
This PR updates the Flash component with the new refresh styles ✨ 

I've also taken this opportunity to refactor a bit. We were previously using the `scheme` prop when we really should have been using `variant`. Sadly, this creates a breaking change but I believe it's worth it to get close to consistency. We were also naming our "schemes" based on colors, when really they should be named to describe the type of Flash (success, warning, danger, default) to aid the user in knowing which variant to use for which purpose.

These new updates also include default styling for icons included in Flash alerts - just like in Primer CSS, we set the color of any icon included in a flash alert to match the variant styling. 

#### Before:
image

#### After:
( I removed the margin between the components to make the code more copy-able, but i'm not 100% attached to this change)
image

### Merge checklist
- [ ] Added or updated TypeScript definitions (`index.d.ts`) if necessary
- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/components/blob/master/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

**Comment by colebemis** [comment]
Is the border-radius of the flash container supposed to be bigger now? It looks the same in the before and after screenshots.

**Comment by emplums** [comment]
Nice catch @colebemis 🤦‍♀️ 😂 

Done @BinaryMuse !

**Comment by BinaryMuse** [comment]
@emplums 🤔🤔🤔 worth using the new deprecation stuff for? (keep `scheme` for now but log a warning and support `variant`)

**Comment by emplums** [comment]
@BinaryMuse yeah good idea! i'll add the deprecation stuff to this PR.

**Comment by emplums** [comment]
Note: @BinaryMuse is going to add a conditional to `deprecate` in a follow up PR to only show the console.warn in development modes

**Comment by emplums** [comment]
@BinaryMuse I _think_ the yarn.lock file in the 19.0.0 branch might be corrupted or something, because I have no changes to package.json in this branch and after merging in the release branch  running `yarn` generates a ton of `yarn.lock` changes 🤔

**Comment by BinaryMuse** [comment]
> @BinaryMuse I _think_ the yarn.lock file in the 19.0.0 branch might be corrupted or something, because I have no changes to package.json in this branch and after merging in the release branch running `yarn` generates a ton of `yarn.lock` changes 🤔

Hm, I pulled the release branch and ran `yarn` but got no changes. Can you share the diff you're getting?

File: index.d.ts [hunk 1]
```diff
@@ -183,6 +183,7 @@ declare module '@primer/components' {
 
   export interface FlashProps extends CommonProps, Omit<React.HTMLAttributes<HTMLDivElement>, 'color'> {
     full?: boolean
+    variant?: 'success' | 'default' | 'warning' | 'danger'
```
Comment (emplums) [inline index.d.ts:186]: https://jsdoc.app/tags-deprecated.html

File: src/Flash.js [hunk 2]
```diff
@@ -1,44 +1,87 @@
...
+      bg: 'flash.danger.bg'
+    },
+    warning: {
+      borderColor: 'flash.warning.border',
+      bg: 'flash.warning.bg'
+    },
+    default: {
+      borderColor: 'flash.default.border',
+      bg: 'flash.default.bg'
+    }
```
Comment (colebemis) [inline src/Flash.js:34]: It looks like we've basically defined the variant structure in the theme. So I think we can replace this object with `scale: 'flash'`:

```suggestion
    scale: 'flash'
```

We'd also have to change `flash.default.border` to `flash.default.borderColor`.

Here's are the styled-system docs about the `scale` option: https://styled-system.com/variants/#themeable-variants

Comment (emplums) [reply src/Flash.js:34]: This doesn't seem to work 🤔

**Pull request merged**